### PR TITLE
mockgcp: support for PubSubSubscription

### PR DESCRIPTION
**Pull request by justinsb**
- **tests: capture golden output for PubSub**
- **mockgcp: support for PubSubSubscription**

### CODE REVIEW

File: mockgcp/mockpubsub/topic.go [hunk 1]
```diff
@@ -48,35 +48,40 @@ func (s *publisherService) CreateTopic(ctx context.Context, req *pb.Topic) (*pb.
...
-			"us-east5",
-			"us-east7",
-			"us-south1",
-			"us-west1",
-			"us-west2",
-			"us-west3",
-			"us-west4",
-			"us-west8",
-		},
+	// TODO: When _is_ this populated?
```
Comment (yuwenma) [inline mockgcp/mockpubsub/topic.go:56]: Good question. I don't know,  but at least in PubSubTopic creation. (I verified against real GCP in the mock storage notification #2447)

File: pkg/test/resourcefixture/testdata/basic/storage/v1beta1/storagenotification/storagenotificationbase/_http.log
Comment (yuwenma) [inline pkg/test/resourcefixture/testdata/basic/storage/v1beta1/storagenotification/storagenotificationbase/_http.log:1]: The real GCP populates the `messageStoragePolicy` in these 2 test suites for storage notification. (The pubSubTopic is a dependency which calls the create method )

**Review by yuwenma — COMMENTED** [review]
/lgtm

you may want to update the populate `MessageStorageePolicy` for PubSubTopic create (at least).

**Comment by justinsb** [comment]
I still haven't found any GCP APIs that are returning the message storage policy... might be an API change.  We should detect it with our e2e tests anyway...

/hold cancel

**Pull request merged**